src/test/java/org/pesmypetcare/webservice/dao/MedicationDaoTest.java

```diff
@@ -0,0 +1,461 @@
+        given(documentSnapshot.exists()).willReturn(true);
+        given(documentSnapshot.get(anyString())).willReturn(value);
+
+        Object medValue = medDao.getMedicationField(owner, petName, dateName, field);
+
+        assertSame(value, medValue, "Should return field value");
+    }
+
+    @Test
+    public void shouldThrowDatabaseAccessExceptionWhenmedDocumentNotExistsInFieldRetrieval() {
+        assertThrows(DatabaseAccessException.class, () -> {
+            given(db.collection(anyString())).willReturn(usersRef);
+            given(usersRef.document(anyString())).willReturn(ownerRef);
+            given(ownerRef.collection(anyString())).willReturn(petsRef);
+            given(petsRef.document(anyString())).willReturn(petRef);
+            given(petRef.collection(anyString())).willReturn(medsRef);
+            given(medsRef.document(anyString())).willReturn(medRef);
+            given(medRef.get()).willReturn(futureDocument);
+            given(futureDocument.get()).willReturn(documentSnapshot);
+            given(documentSnapshot.exists()).willReturn(false);
+
+            medDao.getMedicationField(owner, petName, dateName, field);
+        }, DOCUMENT_NOT_EXISTS_EXC_MSG);
+    }
+
+
+    @Test
+    public void shouldThrowDatabaseAccessExceptionWhenRetrievemedFieldReceivesInterruptedException() {
+        assertThrows(DatabaseAccessException.class, () -> {
+            given(db.collection(anyString())).willReturn(usersRef);
+            given(usersRef.document(anyString())).willReturn(ownerRef);
+            given(ownerRef.collection(anyString())).willReturn(petsRef);
+            given(petsRef.document(anyString())).willReturn(petRef);
+            given(petRef.collection(anyString())).willReturn(medsRef);
+            given(medsRef.document(anyString())).willReturn(medRef);
+            given(medRef.get()).willReturn(futureDocument);
+            willThrow(InterruptedException.class).given(futureDocument).get();
+
+            medDao.getMedicationField(owner, petName, dateName, field);
+        }, INTERRUPTED_EXC_MSG);
+    }
+
+    @Test
+    public void shouldThrowDatabaseAccessExceptionWhenRetrievemedFieldReceivesExecutionException() {
+        assertThrows(DatabaseAccessException.class, () -> {
+            given(db.collection(anyString())).willReturn(usersRef);
+            given(usersRef.document(anyString())).willReturn(ownerRef);
+            given(ownerRef.collection(anyString())).willReturn(petsRef);
+            given(petsRef.document(anyString())).willReturn(petRef);
+            given(petRef.collection(anyString())).willReturn(medsRef);
+            given(medsRef.document(anyString())).willReturn(medRef);
+            given(medRef.get()).willReturn(futureDocument);
+            willThrow(ExecutionException.class).given(futureDocument).get();
+
+            medDao.getMedicationField(owner, petName, dateName, field);
+        }, EXCECUTION_EXC_MSG);
+    }
+
+    @Test
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method length is 17 lines (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5348142)